### PR TITLE
Multiple minor issues

### PR DESCRIPTION
**Pull request by tokee**
A whole bunch of minor issues, all at once because I messed up, sorry. If necessary, I can create separate issues for the improvements listed below:

* Instrumentation has been added to a lot of code parts and the resulting overview has been enhanced with percentages of overall time used.
* The Tika language detection speed improvement [Tika #29](https://github.com/apache/tika/pull/29) has been temporarily copied in order to benefit from the speed without having to use the yet-unreleased Tika 1.8.
* Some trivial speed-improvements were added by replacing String.replaceAll with precompiled Patterns.
* Extraction of meta-data from the ARC path has been added: `ARCNameAnalyser`. The unit test demonstrates how job-names and other data can be extracted.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,5 @@
 build
 target
 credentials.conf
-
+*.iml
+warc-solr-test-server
```
Comment (anjackson) [inline .gitignore]: Why is this in gitignore? The warc-solr-test-server is used during development.

**Comment by anjackson** [comment]
Is it possible to use git rebase or something similar to update this pull request to account for the fact that some of these changes have already been accepted? Or maybe I'm mistaken about that? But one of the tests we just fixed in #52 is still failing here.

**Comment by tokee** [comment]
In theory I could rebase to provide a better overview of issues, but I am scared of that git functionality. I realize that it is a bit confusing to have the same issue addressed multiple times in the timeline. The resulting code is the same though.

**Comment by anjackson** [comment]
Fair enough. I've looked through and it looks good. Can you summarise the change in the CHANGES.md?

**Comment by tokee** [comment]
Thank you for accepting this messy pile. I will endeavor to keep future ones smaller and with a single issue at a time.

**Comment by anjackson** [comment]
No problem. We are hardly paragons of neat coding.

The only problem is that for some weird reason this has broken the CI build: https://travis-ci.org/ukwa/webarchive-discovery/builds/52058185#L5391

**Comment by anjackson** [comment]
Ah, right, so the order of execution of tests is important, and the run order is different on the CI server. Hence, the ThreadLocal Singleton is getting set up earlier on: https://travis-ci.org/ukwa/webarchive-discovery/builds/52119464#L4598

So, could we reset the singleton before the test? Any ideas?

**Comment by tokee** [comment]
I think resetting the singleton is the right solution. That would also allow for extra tests with different configurations in the same test-class.

**Comment by anjackson** [comment]
Any idea how to reset the Singleton?

**Pull request merged**